tests/golden_tests/run_single_files/example.hvm

```diff
@@ -1,25 +1,82 @@
 // Write definitions like this
-(Def1) = (λa a λb b)
+(Def1) = ((λa a) (λb b))
 
 // You can call a definition by just referencing its name
 // It will be substituted in place of the reference
-(Def2) = (λa a Def1)
+(Def2) = ((λa a) Def1)
 
 // Definitions and variables can have names in upper and lower case and contain numbers
 // Names defined in a more inner position shadow names in an outer position
-(def3) = (λDef1 Def1 λx λx x)
+(def3) = ((λDef1 Def1) (λx λx x))
 
 // The language is affine, but if you use a variable more than once the compiler inserts duplications for you
 // Of course you can always do them manually
-(def4) = λz dup z1 z2 = z; (z1 (λx (x x x x x) z2))
+(def4) = λz dup z1 z2 = z; (z1 ((λx (x x x x x)) z2))
 
 // You can use machine numbers and some native numeric operations
 // Numeric operations can only reduce numbers, doing (+ (λx x) 1) will not do anything
 (nums) = λx1 λx2 (* (+ x1 1) (/ (- x2 2) 1))
 
+// You can use numbers on the native match expression
+// The `+` arm binds the `scrutinee`-1 variable to the the value of the number -1
+(Num.pred) = λn 
+  match n { 
+    0: 0
+    +: n-1
+  }
+
+// Write new data types like this
+data Option = (Some val) | None
+data Bool = True | False
+
+// You can have pattern matching on definitions
+// Use `*` to ignore a pattern
+(Option.unwrap_or (Some val) *) = val
+(Option.unwrap_or None      or) = or
+
+(Bool.or True  *) = True
+(Bool.or * True)  = True
+(Bool.or * *)     = False
+
+// Or using a match expression
+(Bool.not) = λbool
+  match bool {
+    True:  False
+    False: True
+  }
+
+// Data types can store values
+data Boxed = (Box val)
+
+// Types with only one constructor can be destructured using `let` or a single matching definition
+(Box.unbox) = λbox
+  let (Box val) = box
+  val
+(Box.map (Box val) f) = (Box (f val))
+
+// Use tuples to store two values together without needing to create a new data type
+(Tuple.new fst snd) =
+  let pair = (fst, snd)
+  pair
+
+// Then you can destructure it inside the definition or using `let`
+(Tuple.fst) = λpair
+  let (fst, snd) = pair
+  fst
+(Tuple.snd (fst, snd)) = snd
```

Comment:
I would change the order and put the pattern matching function first, since it's slightly simpler to understand